medrxivr blogpost

mcguinlu opened the pull request:
Good morning @ropensci/blog-editors, 

Thanks for your patience, as I appreciate this is a day late! I had encountered issues when trying to previewing the post locally with Hugo, but I've checked it using the Netlify build and everything looks as it should.

As this is my second blogpost ever, very happy to incorporate any feedback you have!

Thanks for the opportunity to contribute to the blog,

Luke

steffilazerte commented:
@mcguinlu No worries! I'll take a look at it today and get back to you soon. Thanks!

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,133 @@
+
+### Lack of reproducbility
+
+One of the key tenets of systematic reviews is that literature searches are transparent and reproducible - that is, that a certain search strategy (e.g. "(dementia OR alzheimer) AND lipids") will provide the same number of results each time it is performed. However, anecdotal evidence from information specialists and medical librarians suggests this is not the case for the medRxiv native search:
```
```suggestion
One of the key tenets of systematic reviews is that literature searches are transparent and reproducible - that is, that a certain search strategy (e.g. "(dementia OR alzheimer) AND lipids") will provide the same number of results each time it is performed. However, anecdotal evidence from information specialists and medical librarians suggests that medRxiv's websearch returns inconsistent citations over time.
```

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,133 @@
+{{< tweet 1255485844783325185 >}}
+<!--/html_preserve-->
+
+This means that records returned by a search must be exported by hand in small batches, an error-prone and time-consuming (not to mention anger-inducing!) method of extracting citations from a bibliographic database.
```
Absolutely fine to say "anger-inducing" here :grin:, I'm sure it is!

steffilazerte reviewed (commented):
Hi @mcguinlu 

You said this is your second blog post (congrats)! I really love how awesome the medrxivr package is. It's always great to see a problem solved so elegantly. Your use cases are so lovely and clean and really show case how this package can help so many researchers. I do love my job as an editor :)

In general, most of my comments are stylistic, I think you may have missed some of the check boxes on the [author submission list](https://blogguide.ropensci.org/authorchecklistpeer.html). But those are very quick fixes. The content is well organized, clear and easy to understand. And, as I mentioned above, the use cases are great.

The one thing I'm a bit uncomfortable with is some of the really frustrated tweets. Clearly the interface with medRxiv leaves something to be desired, but I don't want you or rOpenSci to look like you're just bashing medRxiv. I think you illustrate the drawbacks in the text clearly and with grace, without the need for the more negative tweets.

Thanks again for your well written post and useful package!

* [X] post follows Content Guidelines
* [x] post follows Style Guide
* [X] title is in Title Case
* [X] publication date is ok
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok
* [X] YAML package-version included
* [X] YAML subject tags - software peer review, packagename, R; add "community" for post by non-staff non-editor
* [x] acknowledges and links to reviewers
* [x] links to peer review thread

steffilazerte commented:
Oh and if I could get your revisions by Friday or Monday, that'd be great!

mcguinlu commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,133 @@
+
+### Use Case #2 – Perform transparent and reproducible searches as part of a literature review, and export results
+
+While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what `medrxivr` allows you to do once you have imported it. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent. The second core aspect of `medrxivr`'s functionality is designed to address this, allowing information specialis ...
```
So your second interpretation is correct - to search for relevant articles, you first have to import a local copy of the _whole_ database, which is then passed as the `data` argument to `mx_search()`.

I've added some text to this section to try and make this point clearer!

mcguinlu commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,133 @@
+
+## Conclusion
+
+medRxiv is a fantastic resource and has been a key source of information related to the COVID-19 pandemic. However, some key issues with the website mean that its native search functionality precludes its use in systematic reviews. The `medrxivr` R package seeks to address these limitations by providing a user-friendly way to import and systematically search medRxiv and bioRxiv records in R. Full ...
```
I've added some text to explain the NEAR operator, but please let me know if it is still unclear!

mcguinlu commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,133 @@
+
+## Conclusion
+
+medRxiv is a fantastic resource and has been a key source of information related to the COVID-19 pandemic. However, some key issues with the website mean that its native search functionality precludes its use in systematic reviews. The `medrxivr` R package seeks to address these limitations by providing a user-friendly way to import and systematically search medRxiv and bioRxiv records in R. Full ...
```
I've added a linkto the review thread, and added both reviewers names/links to GH profiles to the text.

mcguinlu commented:
Hi @steffilazerte 👋

Thanks a million for your helpful comments - hopefully I have addressed them in my recent commits. I completely agree about the tweets - I just wanted to get across that it's not just me that has encountered these issues and that they are recognised problems in the community, but I can see in retrospect that they come across a bit strong! I've removed them now in favour of the text.

You are right to say I missed a few items on the checklist - in actual fact, I missed the checklist completely! 🤦‍♂️ Very sorry about that, but I think I have addressed all of the items on it now. Just to flag that `roblog::ro_check_url()` was not happy with some of the URLs in the post and I am not entirely sure why. . . . I've reproduced the output below, but just let me know if anything needs fixing:

``` 
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Possibly broken URLs: https://www.medrxiv.org/, https://docs.ropensci.org/medrxivr/, https://www.biorxiv.org/, https://github.com/tts, https://github.com/njahn82, https://github.com/ropensci/software-review/issues/380, https://docs.ropensci.org/medrxivr/, https://doi.org/doi:10.1136/bmj.l2301.
```

steffilazerte commented:
That's great! Thanks for your speedy edits. I'm a bit slow this week (got a cold, bah! With so much social distancing I thought that was impossible!) but I'll hopefully finish the review this afternoon.

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,123 @@
+
+### Use case #2 – Reproducible searching and exporting
+
+While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what medrxivr allows you to do once you have imported a local copy of the database. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent, and while the API (accessed via `mx_api_content()`) is great for accessing medRxiv metad ...
```
```suggestion
While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what medrxivr allows you to do once you have imported a local copy of the database. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent, and while the API (accessed via `mx_api_content()`) is great for accessing medRxiv metadata, it does not offer any search functionality. The second core aspect of medrxivr's functionality is designed to address these limitations, allowing information specialists to build complex searches and apply them to a local copy of the medRxiv database, all while documenting their search strategy in a transparent manner. An example is given below, where a researcher is looking for records that contain both dementia-related and lipid-related terms:
```

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
````diff
@@ -0,0 +1,123 @@
+``` {r}
+
+# Use the maintained snapshot to quickly load today's copy of the medRxiv database
+# Note - this `preprint_data` object could be saved as a CSV in your project repository to aid reproducibility
````
Makes so much more sense!
Small edit to make it fit :smile:

```suggestion
# Note - this `preprint_data` object could be saved as a CSV in your project 
#        repository to aid reproducibility
```

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,123 @@
+# Run your search
+results <- mx_search(data = preprint_data,
+                     query = myquery)
+
```
Perhaps print the results to show the awesome tibble that is returned
```suggestion
results
```

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
````diff
@@ -0,0 +1,123 @@
+
+While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what medrxivr allows you to do once you have imported a local copy of the database. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent, and while the API (accessed via `mx_api_content()`) is great for accessing medRxiv metad ...
+
+``` {r}
````
Below I suggest printing the results, if you like that idea, then I also suggest increasing the width, just a bit to show more of the table
```suggestion
``` {r, R.options=list(width=100)}
```

steffilazerte commented on content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:
```diff
@@ -0,0 +1,123 @@
+
+## Conclusion
+
+medRxiv is a fantastic resource and has been a key source of information related to the COVID-19 pandemic. However, some key issues with the website mean that its native search functionality precludes its use in systematic reviews. The medrxivr R package seeks to address these limitations by providing a user-friendly way to import and systematically search medRxiv and bioRxiv records in R.  Full  ...
```
I had no idea there was such thing as a NEAR operator. That is SO cool

mcguinlu replied:
Yeah, you can currently use it in `medrxivr` via some pretty ugly regex syntax, but implementing it in a nicer fashion is definitely on the horizon!

steffilazerte reviewed (commented):
This looks great! I have one typo fix and two suggestions for the final code block (feel free to take or leave them). 
Thanks for your 🚀  work

mcguinlu commented:
Great, thanks! Edits committed, plus one small final change from me (short comment in the code block just before the `results` object is printed). 

 Wishing you a very nice weekend, and a speedy recovery from the cold!

steffilazerte commented:
Awesome, we're good to go! Thank you for your post, I'll publish and tweet on Tuesday 😸

stefaniebutland commented:
@mcguinlu I just noticed that in the post, the package itself isn't mentioned until "Use cases". Would you consider adding a sentence or two right before Background so people know what the post and package are about? Maybe consider adding to the title? Searching medRxiv and bioRxiv Preprint Data with the medrxivr package.

cc @steffilazerte

steffilazerte commented:
@mcguinlu That's right, I completely missed that you removed a paragraph from the beginning, definitely consider adding it back in, I quite liked it 😄 

```
However, the native search functionality available on the medRxiv website is not suitable for
use in systematic reviews, due to a number of limitations. These limitations, presented below,
prompted the development of [medrxivr](https://docs.ropensci.org/medrxivr) R package which 
provides access to, and tools for searching, medRxiv preprint metadata. This post will also
summarize the key functionality of the package with respect to two key use cases. Note that 
medrxivr allows users to access and search bioRxiv as well as medRxiv preprint metadata, 
as both repositories are run by the Cold Spring Harbor Laboratory and so have a similar 
API endpoint/native website search functionality, but all illustrating examples in this post are
taken from medRxiv.
```

mcguinlu commented:
Whoops - not entirely sure how I managed to delete that paragraph, but have re-added it now! Great catch @stefaniebutland!
